langchain/agents/tools/sql.ts

```diff
@@ -0,0 +1,217 @@
+}
+
+export class InfoSqlTool extends Tool implements SqlTool {
+  name = "info-sql";
+
+  db: SqlDatabase;
+
+  constructor(db: SqlDatabase) {
+    super();
+    this.db = db;
+  }
+
+  // get the table names for a comma separated list of tables, return error message in strong format
+  async call(input: string) {
+    try {
+      const tables = input.split(",").map((table) => table.trim());
+      const createTableStatements = await Promise.all(
+        tables.map((table) => this.db.getCreateTableStatement(table))
+      );
+      const sampleData = await Promise.all(
+        tables.map((table) => this.db.getSampleData(table))
+      );
+      return tables
+        .map(
+          (_, index) =>
+            `${createTableStatements[index]}\n${sampleData[index].join("\n")}`
+        )
+        .join("\n\n");
+    } catch (error) {
+      return `${error}`;
+    }
+  }
+
+  description = `Input to this tool is a comma-separated list of tables, output is the schema and sample rows for those tables.
+    Be sure that the tables actually exist by calling list_tables_sql_db first!
+    
+    Example Input: "table1, table2, table3.`;
+}
+
+export class ListTablesSqlTool extends Tool implements SqlTool {
+  name = "list-tables-sql-db";
+
+  db: SqlDatabase;
+
+  constructor(db: SqlDatabase) {
+    super();
+    this.db = db;
+  }
+
+  async call(_: string) {
+    try {
+      const tables = await this.db.getTables();
+      return tables.sort().join("\n");
+    } catch (error) {
+      return `${error}`;
+    }
+  }
+
+  description = `Input is an empty string, output is a comma separated list of tables in the database.`;
+}
+
+export class QueryCheckerTool extends Tool {
+  name = "query-checker";
+
+  template = `
+    {query}
+Double check the sqlite query above for common mistakes, including:
+- Using NOT IN with NULL values
+- Using UNION when UNION ALL should have been used
+- Using BETWEEN for exclusive ranges
+- Data type mismatch in predicates
+- Properly quoting identifiers
+- Using the correct number of arguments for functions
+- Casting to the correct data type
+- Using the proper columns for joins
+
+If there are any of the above mistakes, rewrite the query. If there are no mistakes, just reproduce the original query.`;
+
+  llmChain: LLMChain;
+
+  constructor(llmChain?: LLMChain) {
+    super();
+    if (llmChain) {
+      this.llmChain = llmChain;
+    } else {
+      const model = new OpenAI({ temperature: 0 });
+      const prompt = new PromptTemplate({
+        template: this.template,
+        inputVariables: ["query"],
+      });
+      this.llmChain = new LLMChain({ llm: model, prompt });
+    }
+  }
+
+  async call(input: string) {
+    return this.llmChain.predict({ query: input });
+  }
+
+  description = `
+    Use this tool to double check if your query is correct before executing it.
+    Always use this tool before executing a query with query_sql_db!`;
```

Comment:
This seems to be referencing the sql tool by a different name than it has in its definition

Reply:
good catch